consensus/src/reporter.rs

```diff
@@ -20,17 +21,30 @@ impl<A: Send, R: Reporter<Activity = A>> Reporter for Option<R> {
 }
 
 /// A struct used to report activity to multiple [Reporter]s (which may or may not be present).
-#[derive(Clone)]
 pub struct Reporters<A, R1, R2> {
     r1: Option<R1>,
     r2: Option<R2>,
 
     _phantom: PhantomData<A>,
 }
 
+impl<A, R1, R2> Clone for Reporters<A, R1, R2>
+where
+    R1: Clone,
+    R2: Clone,
+{
+    fn clone(&self) -> Self {
```

Comment:
Suspect this complains for the same reason you mentioned below around A not being clone?

Reply:
Yeah. Derive macros essentially do something like `Clone for MyType<A> where A: Clone`, but `A: !Clone` almost always.

Another one of those things that'd be nice to have: https://github.com/rust-lang/lang-team/issues/152